src/Stratis.Bitcoin/P2P/Peer/NetworkPeerFactory.cs

```diff
@@ -70,11 +70,11 @@ public interface INetworkPeerFactory
         /// <para>When created, the server is ready to be started, but this method does not start listening.</para>
         /// </summary>
         /// <param name="network">Specification of the network the node runs on - regtest/testnet/mainnet.</param>
-        /// <param name="localEndPoint">IP address and port to listen on, or <c>null</c> to listen on all available interfaces and default port.</param>
-        /// <param name="externalEndPoint">IP address and port that the server is reachable from the Internet on, or <c>null</c> to use the same value as <paramref name="localEndPoint"/>.</param>
+        /// <param name="localEndPoint">IP address and port to listen on.</param>
+        /// <param name="externalEndPoint">IP address and port that the server is reachable from the Internet on.</param>
         /// <param name="version">Version of the network protocol that the server should run.</param>
         /// <returns>Newly created network peer server, which is ready to be started.</returns>
-        NetworkPeerServer CreateNetworkPeerServer(Network network, IPEndPoint localEndPoint = null, IPEndPoint externalEndPoint = null, ProtocolVersion version = ProtocolVersion.PROTOCOL_VERSION);
+        NetworkPeerServer CreateNetworkPeerServer(Network network, IPEndPoint localEndPoint, IPEndPoint externalEndPoint, ProtocolVersion version = ProtocolVersion.PROTOCOL_VERSION);
```

Comment:
Is that the fix for the bug where a node would connect to itself?

Reply:
No, there was a wrong merge once and due to these default values it was unnoticed. So it was suggested that we remove the default values because we don't use them anyway. In my view, the default values did make sense and the initialization of such object was reasonable, but it is true that currently, it is of no use, so I was OK to remove them.

Reply:
cool thanks